ignite/pkg/localfs/watcher.go

```diff
@@ -97,6 +104,9 @@ func (w *watcher) listen() {
 	for {
 		select {
 		case e := <-w.wt.Event:
+			if e.IsDir() && w.ignoreFolders {
```

Comment:
I added this, because I realized that ignored files still trigger rebuilds in chain serve on the folder level. Example scenario where the build triggered a new build:
1. Delete *.pg.* files from x/module/types
2. ignite c serve
3. When the the new pg files are created, the watcher gets triggered on the x/module/types folder, which triggers another build

I think this sort of worked before, but with the cacheStorage (bolt) lock this caused problems. Ignoring folder changes seems fairly safe to me since I assume we actually just want file changes?

Reply:
There is a case that I think might be a problem:
- Create a new pkg under app/
- Import that in app.go - this will trig rebuild
- Modify the package we created in the first step - it should rebuild but it won't

Reply:
Im not sure I understand. Would the new files under the new pkg not get picked up?

Reply:
Yes, the files in the new package won't get picked up, they won't trigger a rebuild.

Reply:
I figured out what caused this. It was the limit of single events. When the folder was ignored the other file events was ignored. I solved this by changing the way we watch for files. Instead of watching everything, we can use a filter to avoid watching things we are supposed to ignore. That solves this pretty neatly.